docs/upgrading.md

```diff
@@ -1,66 +1,104 @@
+
+## Changes in 0.15
+
+Contour 0.15 has made a change to the deployment manifests which you should be aware of.
+
+### Breaking change
+
+Contour's `contour serve` now requires that either TLS certificates be available, or
+you supply the `--insecure` parameter.
+
+**If you do not supply TLS details or `--insecure`, `contour serve` will not start.**
 
 ## The less easy way
 
 ### Required changes
 
-You *must* either enable TLS for gRPC serving, or put `--insecure` into your `contour serve` startup line. If you are running with both Contour and Envoy in a single pod, the existing deployment examples have already been updated with this change.
+You *must* either enable TLS for gRPC serving, or put `--insecure` into your `contour serve`
+startup line. If you are running with both Contour and Envoy in a single pod, the existing
+deployment examples have already been updated with this change.
+
+If you are running using the `ds-hostnet-split` example or a derivative, we strongly recommend
+that you generate new certificates for securing your gRPC communication between Contour and Envoy.
+
+### Other changes
+
+The other major feature that is available is an experimental version of Leader Election.
+Currently, you can enable leader election with the `--enable-leader-election` flag to
+`contour serve`. This will ensure that *only the elected leader serves the gRPC endpoints.*
+That is, only the elected leader will actually open the gRPC port and start serving.
+
+If you have deployed Contour and Envoy in their own pods - we call this split deployment - you
+should enable leader election so all envoy pods take their configuration from the lead contour. 
 
-If you are running using the `ds-hostnet-split` example or a derivative, we strongly recommend that you generate new certificates for securing your gRPC communication between Contour and Envoy.
+There are some further requirements for this, see the [howto](#enabling-leader-election) section below for more.
 
-There is a Job in the `ds-hostnet-split` directory that will use the new `contour certgen` command to generate a CA and then sign Contour and Envoy keypairs, which can also then be saved directly to Kubernetes as Secrets, ready to be mounted into your Contour and Envoy Deployments and Daemonsets.
+### Enabling TLS for gRPC
+
+There is a Job in the `ds-hostnet-split` directory that will use the new `contour certgen` command
+to generate a CA and then sign Contour and Envoy keypairs, which can also then be saved directly
+to Kubernetes as Secrets, ready to be mounted into your Contour and Envoy Deployments and Daemonsets.
 
 Again, check the examples for Contour and Envoy in `ds-hostnet-split` for the exact commands.
 
-The steps you will need to follow for this to work are detailed more in (grpc-tls-howto.md)[./grpc-tls-howto.md]:
+If you wish to do this manually, you can use the steps detailed in (grpc-tls-howto.md)[./grpc-tls-howto.md].
+
+In short, you will need to:
```

Comment:
If this duplicates grpc-tls-howto.md)[./grpc-tls-howto.md, then dont' list it here to avoid information skew